code/lib/preview-api/src/modules/store/csf/portable-stories.ts

```diff
@@ -244,27 +244,73 @@ export function createPlaywrightTest<TFixture extends { extend: any }>(
             `);
         }
 
+        // start the play function on the client, and halt when rendering starts
         await page.evaluate(async (wrappedStoryRef: WrappedStoryRef) => {
           const unwrappedStoryRef = await globalThis.__pwUnwrapObject?.(wrappedStoryRef);
           const story =
             '__pw_type' in unwrappedStoryRef ? unwrappedStoryRef.type : unwrappedStoryRef;
-          return story?.load?.();
+
+          const renderingStarted = Promise.withResolvers<void>();
+          story.renderingEnded = Promise.withResolvers();
+          story.playPromise = story.play({
+            canvasElement: document.querySelector('#root'),
+            renderToCanvas: async () => {
+              renderingStarted.resolve();
+              await story.renderingEnded?.promise;
+            },
+          });
+          await renderingStarted.promise;
         }, storyRef);
 
-        // mount the story
+        // let playwright mount the story in node
         const mountResult = await mount(storyRef, ...restArgs);
 
-        // play the story in the browser
+        // go back to client to continue playing the play function
         await page.evaluate(async (wrappedStoryRef: WrappedStoryRef) => {
           const unwrappedStoryRef = await globalThis.__pwUnwrapObject?.(wrappedStoryRef);
           const story =
             '__pw_type' in unwrappedStoryRef ? unwrappedStoryRef.type : unwrappedStoryRef;
-          const canvasElement = document.querySelector('#root');
-          return story?.play?.({ canvasElement });
+
+          story.renderingEnded?.resolve();
+          await story.playPromise;
         }, storyRef);
 
         return mountResult;
       });
     },
   });
 }
+
+// TODO At some point this function should live in prepareStory and become the core of StoryRender.render as well.
+// Will make a follow up PR for that
```

Comment:
Reminder to create an issue for this followup

Reply:
done